azerothcore/azerothcore-wotlk
# fix(core/gossip): Replace hardcoded string of vendors and trainers with gossip menu item

## Changes Proposed:
-  GOSSIP_TEXT_BROWSE_GOODS and GOSSIP_TEXT_TRAIN replaced wifh gossip menu items
-  Added condition to see menu items from GM-mode

## Issues Addressed:

- Closes some localisation issues

## SOURCE:


## Tests Performed:

- Tested on AzerothCore rev. 1f90a18a5888


## How to Test the Changes:


1. `.go c 97713`
In GM-mode:

![изображение](https://user-images.githubusercontent.com/25766571/157981328-3aa1e93b-fa3c-4bcc-8a5d-8b8df949d0b1.png)

If GM-mode is OFF:

![изображение](https://user-images.githubusercontent.com/25766571/157981482-b71e5684-9332-4335-bbea-68443039064a.png)

2. `.go c 42201` and other trainers or vendors

In GM-mode

![изображение](https://user-images.githubusercontent.com/25766571/157981949-648b264f-eecb-43ba-a19f-86c6788c9d06.png)

if GM-mode is OFF:

![изображение](https://user-images.githubusercontent.com/25766571/157982134-1c67422b-1c41-4ebc-ae33-bdc4c1040132.png)

If patch is incorrect - you'll not see "Train me!" or "I'd like to browse your goods." 
If correct - you'll see them or your localized strings.


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
